### PR TITLE
Add unit tests for pooler activation functions

### PR DESCRIPTION
**Pull request by taneem-ibrahim**
## Purpose

Add dedicated unit tests for the pooler activation classes in `vllm/model_executor/layers/pooler/activations.py`, which previously had zero unit test coverage.

## Test Plan
- `python -m pytest tests/model_executor/layers/test_pooler_activations.py -v`
- `pre-commit run ruff-check --files tests/model_executor/layers/test_pooler_activations.py`
 - `pre-commit run ruff-format --files tests/model_executor/layers/test_pooler_activations.py`

## Test Result
`================== 29 passed, 16 warnings in 4.61s ==========================`
`ruff check...............................................................Passed`
`ruff format..............................................................Passed`

### CODE REVIEW

File: tests/model_executor/layers/test_pooler_activations.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    def test_static_reads_config(self, vllm_config):
+        pooler = PoolerClassify(static_num_labels=True)
+        assert pooler.num_labels is not None
+
+    def test_static_default_num_labels_zero_uses_sigmoid(self, vllm_config):
+        pooler = PoolerClassify(static_num_labels=True)
+        assert pooler.num_labels == 0
+        x = torch.zeros(1, 3)
+        out = pooler(x)
+        assert torch.allclose(out, torch.full((1, 3), 0.5), atol=1e-5)
```
Comment (yewentao256) [inline tests/model_executor/layers/test_pooler_activations.py:124]: Assertion for sigmoid, but the warning suggests for softmax in `vllm-source/vllm/model_executor/layers/pooler/activations.py`

```py
if num_labels == 0:
            logger.warning(
                "num_labels should be > 0 for classification "
                "models, falling back to softmax. "
                "Please check if the configuration is correct."
            )
```

File: tests/model_executor/layers/test_pooler_activations.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    def test_dynamic_sigmoid_when_num_labels_lt_2(self):
+        pooler = PoolerClassify(static_num_labels=False)
+        x = torch.zeros(1, 1)
+        out = pooler(x)
+        assert torch.allclose(out, torch.tensor([[0.5]]), atol=1e-5)
+
+    def test_static_reads_config(self, vllm_config):
+        pooler = PoolerClassify(static_num_labels=True)
+        assert pooler.num_labels is not None
+
```
Comment (yewentao256) [inline tests/model_executor/layers/test_pooler_activations.py]:
```suggestion
```
This unit test is not needed as we already fixture `num_labels=0`

File: tests/model_executor/layers/test_pooler_activations.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    def test_wraps_softmax(self, vllm_config):
+        result = PoolerActivation.wraps(nn.Softmax(dim=-1))
+        assert isinstance(result, PoolerClassify)
+
+    def test_wraps_other_module(self):
+        result = PoolerActivation.wraps(nn.ReLU())
+        assert isinstance(result, LambdaPoolerActivation)
+
+
```
Comment (yewentao256) [inline tests/model_executor/layers/test_pooler_activations.py]:
```suggestion
```

This only covers for isinstance, not that meaningful, I think can be deleted as well.

**Review by yewentao256 — COMMENTED** [review]
Thanks for the work!

File: tests/model_executor/layers/test_pooler_activations.py (hunk 2 above)
Comment (taneem-ibrahim) [reply tests/model_executor/layers/test_pooler_activations.py]: Good catch. Removed this one.

File: tests/model_executor/layers/test_pooler_activations.py (hunk 1 above)
Comment (taneem-ibrahim) [reply tests/model_executor/layers/test_pooler_activations.py:124]: Updated the warning message to `sigmoid`.  Thank you for catching this!

File: tests/model_executor/layers/test_pooler_activations.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        cfg = self._make_config(problem_type="regression")
+        result = get_act_fn(cfg)
+        assert isinstance(result, PoolerIdentity)
+
+    def test_single_label_classification(self, vllm_config):
+        cfg = self._make_config(
+            problem_type="single_label_classification", num_labels=3
+        )
+        result = get_act_fn(cfg)
+        assert isinstance(result, PoolerClassify)
```
Comment (yewentao256) [inline tests/model_executor/layers/test_pooler_activations.py:173]:
```suggestion
    def test_single_label_classification(self, vllm_config):
        vllm_config.model_config.hf_config.num_labels = 3
        cfg = self._make_config(problem_type="single_label_classification")

        result = get_act_fn(cfg)
        assert isinstance(result, PoolerClassify)
```
Here is tricky, 
```py
class PoolerClassify(PoolerActivation):
    def __init__(self, *, static_num_labels: bool = True) -> None:
        super().__init__()

        if static_num_labels:
            vllm_config = get_current_vllm_config()
            model_config = vllm_config.model_config
            num_labels = getattr(model_config.hf_config, "num_labels", 0)
        else:
            num_labels = None
```
static_num_labels is passed but reading from hf_config, would recommend taking a look and have a following up PR fixing this.

**Review by yewentao256 — APPROVED** [review]
LGTM, thanks for the work!

File: tests/model_executor/layers/test_pooler_activations.py (hunk 4 above)
Comment (taneem-ibrahim) [reply tests/model_executor/layers/test_pooler_activations.py:173]: Thanks, Wentao. I will work on a new PR to resolve this one.
